tests/scripts/all.sh

```diff
@@ -823,6 +823,16 @@ pre_generate_files() {
     fi
 }
 
+check_test_extended () {
+    # Usage: put "check_test_extended || return" at the top of every test that should
+    # only run when extended tests are enabled. Doing this instead of using a
+    # supports_xxx function stops the CI from treating this as a not-tested component.
+    if [ -z ${MBEDTLS_TEST_EXTENDED+x} ]; then
```

Comment:
Please don't do that. If the CI says that `test_valgrind` passed, I want it to mean that it passed, not that it didn't run. Otherwise we lose track of which tests are actually running (which has been a problem in various ways in the past).

Reply:
> I'm going to take the hacky solution for now and we can make a cleaner solution later.

Sorry, no, that's not good. Please use a solution that doesn't lie to us.

It doesn't even have to be more complicated than what you're doing now. You just need to tweak the groovy scripts instead of `all.sh`. A hacky solution is a simple patch to the Groovy scripts.

Reply:
The Groovy patch would be to discard certain components from `all_all_sh_components` in `get_branch_information` [in `common.groovy`](https://github.com/Mbed-TLS/mbedtls-test/blob/daeca6497730839f83b86bb9ee28968225f1ef75/vars/common.groovy#L277). Something like:

```
if (job_is_pr_job) {
    all_all_sh_components.removeAll {k,_ -> k ==~ /valgrind/}
}
```

Reply:
Actually, I prefer having `all.sh` components be able to say whether they should be run in all builds or only full builds. That's a more extensible approach than just saying "skip all valgrind tests". Sure, we're doing this for valgrind, but the optimal latency/set-of-tests-run point will differ for PR builds vs merge queue builds (would be great if we could target these!) vs nightlies or even weeklies

Reply:
> I prefer having all.sh components be able to say whether they should be run in all builds or only full build

So do I, but I don't see a way to do that quickly. It's the way I'd do it if this task was on the tech debt pile. As an urgent patch, the only way I can think of is to let Groovy decide.

Reply:
Actually, while I don't see quick a way to _do this from the mbedtls repository alone_, I do see a quick way of _letting the mbedtls repository decide_.

* In `all.sh`, add an option `--list-pr-components` that is like `--list-all-components`, but omits some.
* In `common.groovy`, if `--list-pr-components` is found in the `--help` text and we're doing a PR job, use that. Otherwise (older mbedtls branch or release job), use the output of `--list-all-components` as is done today.

Reply:
I suggest that we separate the components into `component_xxx` and `component_release_xxx` and filter the components in Groovy to only run the release components in nightly and release jobs.

Reply:
Please take a look at this mbedtls-test PR: https://github.com/Mbed-TLS/mbedtls-test/pull/143